Tutorials/supervised.ipynb

```diff
@@ -0,0 +1,373 @@
+    "        # only pick zero-magnetization states\n",
+    "        mag = np.sum(visible)\n",
+    "        if(np.abs(mag) < 1.0e-4):\n",
+    "            tsamples.append(visible.tolist())\n",
+    "            ttargets.append([np.log(res.eigenvectors[0][i])])\n",
+    "\n",
+    "    return hi, tsamples, ttargets\n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "After we obtain the result as ``res``, we return the hilbert space ``hi``, the spin basis ``tsamples``, and the coefficients ``ttargets``.\n",
+    "\n",
+    "Notice that we restrict ourselves to $\\sum S_z = 0$ symmetry sector to simplify the learning."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "We now consider a small system $L=10$ and with $J_2 = 0.4$, and obtain the data by calling the function ```load_ed_data```."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 3,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "L = 10\n",
+    "J2 = 0.4\n",
+    "\n",
+    "# Load the Hilbert space info and data\n",
+    "hi, training_samples, training_targets = load_ed_data(L, J2)\n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## 2) Choosing the Machine and the Optimizer\n",
+    "\n",
+    "For this tutorial, we consider the Restricted Bolzmann Machine ``nk.machine.RbmSpin`` and the AdaDelta optimizer ``nk.optimizer.AdaDelta``."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 4,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Machine\n",
+    "ma = nk.machine.RbmSpin(hilbert=hi, alpha=1)\n",
+    "ma.init_random_parameters(seed=1234, sigma=0.01)\n",
+    "# Optimizer\n",
+    "op = nk.optimizer.AdaDelta()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## 3) Defining the Supervised Learning object\n",
+    "\n",
+    "We have now have almost everything (machine, optimizer, data) for setting up a supervised learning object. We also need to provide the batch size, ``batch_size``, for stochatic gradient descent. For detail, see https://en.wikipedia.org/wiki/Stochastic_gradient_descent\n",
+    "\n",
+    "\n"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 5,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Supervised learning object\n",
+    "spvsd = nk.supervised.Supervised(\n",
+    "    machine=ma,\n",
+    "    optimizer=op,\n",
+    "    batch_size=400,\n",
+    "    samples=training_samples,\n",
+    "    targets=training_targets)\n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## 4) Running the Supervised Learning\n",
+    "\n",
+    "The very last piece we need for supervised learning is the loss function.\n",
+    "\n",
+    "### Loss function\n",
+    "There are different loss functions one could define for the optimization problem, for example:\n",
+    "\\begin{align*}\n",
```

Comment:
There is some formatting issues with these lines of LateX, maybe some missing $$ ?